jumanji/environments/combinatorial/cvrp/specs_test.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+import chex
+import jax.numpy as jnp
+import pytest
+
+from jumanji.environments.combinatorial.cvrp.env import CVRP
+from jumanji.environments.combinatorial.cvrp.types import Observation
+
+
+class TestObservationSpec:
+    env = CVRP()
+    observation_spec = env.observation_spec()
+    observation = observation_spec.generate_value()
+
+    def test_observation_spec__generate_value(self) -> None:
+        """Test generating a value which conforms to the observation spec."""
+        assert isinstance(self.observation, Observation)
+
+    def test_observation_spec__validate(self) -> None:
+        """Test the validation of an observation given the observation spec."""
+        observation = self.observation_spec.validate(self.observation)
+        # Check that a different shape breaks the validation
+        with pytest.raises(ValueError):
+            modified_shape_observation = observation._replace(
+                problem=observation.problem[None, ...]
+            )
+            self.observation_spec.validate(modified_shape_observation)
```

Comment:
The validation should raise an error but not the `_replace`, so I think it is better to move the replace up.
```suggestion
        modified_shape_observation = observation._replace(
            problem=observation.problem[None, ...]
        )
        with pytest.raises(ValueError):
            self.observation_spec.validate(modified_shape_observation)
```